### PR TITLE
chore: add number field field spec

### PR DESCRIPTION
**Pull request by chrisdholt**
# Description
Adds a new specification for number field. 

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: specs/number-field.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+- `list` - the id of the `<datalist>` element that contains the optional pre-defined autocomplete options
+- `max`	- the maximum value to accept for this input
+- `min` -	the minimum number to accept for this input
+- `maxlength`	- the maximum number of characters the input should accept
+- `minlength` -	the minimum number of characters long the input can be and still be considered valid
+- `name` - the name of the control
+- `size` - a number indicating how many characters wide the input field should be
+- `appearance` - enum
+  - outline (default)
+  - filled
```
Comment (EisenbergEffect) [inline specs/number-field.md]: Should we also include whether or not the number can contain a decimal and how many decimal places it supports?

Comment (EisenbergEffect) [reply specs/number-field.md]: Should we support a `step` property that indicates how much to increment/decrement by? As an example, imagine an audio app that wants to represent tuning in KHz. They might want to have an increment of .1 KHz.

File: specs/number-field.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+### States
+
+**disabled**: `true` or `false` - when disabled, the value will not be changeable through user interaction. It should also not expose it's value to a form submission.
+
+**readonly**: `true` or `false` - when readonly, the value will not be changeable through user interaction. The value will still be exposed to forms on submission.
+
+### Accessibility
+
+The host element will be the focusable element and should handle providing the role of textbox to assistive technology (AT) as well as reporting any state based aria attributes (disabled, readonly, etc) to AT.
```
Comment (EisenbergEffect) [inline specs/number-field.md]: Are there any keyboard interactions for increment/decrement of the number?

File: specs/number-field.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+*Screenshots below are of the basic appearance of the component and are not exhaustive.*
+
+**The visual treatment of this control should visually match the number field component**
+
+| State | Image |
+| ----- | ----- |
+| default | ![](../text-field/images/text-field.png) |
+| focus | ![](../text-field/images/text-field-focus.png)
+| filled | ![](../text-field/images/text-field-filled.png)
```
Comment (EisenbergEffect) [inline specs/number-field.md]: Images didn't seem to show up for me.

File: specs/number-field.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+- disabled
+- required
+- readonly
+
+*CSS Parts*
+- label
+- root
+- start
+- end
+- control
```
Comment (EisenbergEffect) [inline specs/number-field.md]: Would we provide custom spin buttons with the ability to style them and replace their icon?

File: specs/number-field.md (hunk 1 above)
Comment (chrisdholt) [reply specs/number-field.md]: I think step for sure should be added, I'll update with that.

File: specs/number-field.md (hunk 2 above)
Comment (chrisdholt) [reply specs/number-field.md]: Sure, I can articulate that.

File: specs/number-field.md (hunk 3 above)
Comment (chrisdholt) [reply specs/number-field.md]: Odd, I'll see if I can figure out why...or I'll dupe and rename :)

File: specs/number-field.md (hunk 4 above)
Comment (chrisdholt) [reply specs/number-field.md]: I think we'll provide slots for it, but I think the buttons themselves might be a different spec and PR.

Comment (chrisdholt) [reply specs/number-field.md]: Initially I think the spin buttons could be passed into the existing `end` slot. I'll add next steps for creating those and passing them by default.

File: specs/number-field.md (hunk 1 above)
Comment (chrisdholt) [reply specs/number-field.md]: added step 👍

**Pull request merged**